model/testing/src/icon4py/model/testing/stencil_tests.py

```diff
@@ -77,29 +82,56 @@ def test_and_benchmark(
     grid: base.Grid,
     _properly_allocated_input_data: dict[str, gtx.Field | tuple[gtx.Field, ...]],
     _configured_program: Callable[..., None],
+    request: pytest.FixtureRequest,
 ) -> None:
-    reference_outputs = self.reference(
-        _ConnectivityConceptFixer(
-            grid  # TODO(havogt): pass as keyword argument (needs fixes in some tests)
-        ),
-        **{
-            k: v.asnumpy() if isinstance(v, gtx.Field) else v
-            for k, v in _properly_allocated_input_data.items()
-        },
-    )
-
-    _configured_program(**_properly_allocated_input_data, offset_provider=grid.connectivities)
-    self._verify_stencil_test(
-        input_data=_properly_allocated_input_data, reference_outputs=reference_outputs
-    )
+    benchmark_only_option = request.config.getoption("benchmark_only")
+    benchmark_only_mark = request.node.get_closest_marker("benchmark_only") is not None
+    if (not benchmark_only_option) and (not benchmark_only_mark):
+        reference_outputs = self.reference(
+            _ConnectivityConceptFixer(
+                grid  # TODO(havogt): pass as keyword argument (needs fixes in some tests)
+            ),
+            **{
+                k: v.asnumpy() if isinstance(v, gtx.Field) else v
+                for k, v in _properly_allocated_input_data.items()
+            },
+        )
+
+        _configured_program(**_properly_allocated_input_data, offset_provider=grid.connectivities)
+        self._verify_stencil_test(
+            input_data=_properly_allocated_input_data, reference_outputs=reference_outputs
+        )
 
     if benchmark is not None and benchmark.enabled:
+        warmup_enabled = request.config.getoption("benchmark_warmup")
+        if warmup_enabled:
+            print("[WARNING] Benchmark warmup enabled, GT4Py timers include warmup iterations.")
```

Comment:
What's the default warmup behavior here?

Reply:
The warmup is done from `pytest-benchmark`. There's no specific amount of warmup iterations IIRC. The warmup iterations depend on the duration of the test. I didn't want to complicate things more and reimplement the warm up so I thought at least we can add a warning to explain why the number of repetitions of the test in the pytest results differs from the repetitions reported by the gt4py timers. If you think it's confusing I can remove the warning

Reply:
We could disable the pytest-benchmark warmup and do it ourselves or use the pedantic mode https://pytest-benchmark.readthedocs.io/en/latest/pedantic.html to use a fixed number of warmup steps.